### DISCUSSION TITLE
Is Socks5 inbound via UDS possible?

### DISCUSSION THREAD

**Discussion by GektorUA** [XTLS/Xray-core#1607] [General]
Is it possible configure Xray Socks5 (TCP+UDP) inbound via Unix Domain Socket? How to do it?

**Comment by wulfric93** [comment]
its my question too

**Comment by us254** [comment]


**Reply by wulfric93** [reply]
can you explain more ?

**Comment by wulfric93** [comment]
there is no socks.sock file available 
<img width="247" alt="image" src="https://github.com/XTLS/Xray-core/assets/117269334/0db07bf5-1209-452c-b375-8c4364745403">

**Reply by GektorUA** [reply]
i have try Xray-core (much older then 1.8.4 version) configure to use Socks5 (TCP+UDP) inbound via Unix Domain Socket on OpenWrt, but unsuccessful, Xray did not create any file.

**Comment by GektorUA** [comment]
There is explanation of UDS by XRay:
https://xtls.github.io/config/transports/domainsocket.html#domainsocketobject

**the transmission method can be TCP, WebSocket, HTTP/2**

Seems Socks5 is not supported.

**Comment by us254** [comment]


**Reply by GektorUA** [reply]
Nope, it'd did not work:

```
2023/10/14 15:37:36 [Info] infra/conf/serial: Reading config: config.json
Failed to start: main: failed to load config files: [config.json] > infra/conf: Config: unknown transport protocol: unix
```

**Comment by us254** [comment]
```json
"inbounds": [
  {
    "protocol": "socks",
    "settings": {
      "udp": true,
      "unix": "/path/to/your/unix.sock"
    }
  }
]
```
